src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -264,10 +311,12 @@ public Response downloadZip(JsonObject content,
             Result result = controller.execute();
             if (result instanceof Failed)
             {
-               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(result.getMessage()).build();
+               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(getMessage(result)).build();
```

Comment:
I think it should be better if we return a JSON with the `org.jboss.forge.service.util.UICommandHelper.describeResult(JsonObjectBuilder, Result)` messages instead